### PR TITLE
Gp2-3102: Vulnerable JS packages in great-cms/domestic/\* need updating

### PR DESCRIPTION
**Pull request by seijihg**
Updated packages and gulpfile.js in both core/components and domestic. Updated commands to make it more streamlined in both places. Everything is updated in the README.md

### Workflow

- [x] Ticket exists in Jira https://uktrade.atlassian.net/browse/GP2-3102
- [x] Jira ticket has the correct status.
- [x] [Changelog](CHANGELOG.md) entry added.

### Reviewing help

- [ ] Explains how to test locally, including how to set up appropriate data
- [ ] Includes screenshot(s) - ideally before and after, but at least after

### Housekeeping

- [ ] Added all new environment variables to Vault.
- [ ] Cleaned up old feature flags
- [ ] Upgraded any vulnerable dependencies.
- [ ] Ran the `make manage download_geolocation_data` command
- [ ] I have updated security dependencies
- [ ] Python requirements have been re-compiled.
- [x] Frontend assets have been re-compiled.

### Merging

- [x] This PR can be merged by reviewers. (If unticked, please leave for the author to merge)

### CODE REVIEW

File: core/components/static/stylesheets/elements-components.css [hunk 1]
```diff
@@ -1301,7 +1301,8 @@ a.labelled-image-card {
 .card, a.labelled-card, a.labelled-image-card {
   margin-bottom: 30px;
   width: 100%;
-  background: #ffffff; }
+  background: #ffffff;
+  background-color: blue; }
```
Comment (stevejalim) [inline core/components/static/stylesheets/elements-components.css:1305]: When i run gulp build in core/components, this change is removed

```
-  background: #ffffff;
-  background-color: blue; }
+  background: #ffffff; }
```

Maybe worth re-running gulp build on this branch and seeing if it disappears for you, too? If so, check in the 'reverted' file, I think...

**Review by stevejalim — COMMENTED** [review]
LGTM - got a comment about a file that maybe shouldn't have changed / maybe changes back if you re-run the command.

Also, lots of deprecated package warnings, but it DOES work, which is good :)

Hopefully none of the new packages will have theirown vulns!

**Comment by stevejalim** [comment]
Important postscript: NICE WORK. Thank you! 🙌

File: core/components/gulpfile.js.bak [hunk 2]
```diff
@@ -0,0 +1,169 @@
+'use strict'
```
Comment (bobmeredith) [inline core/components/gulpfile.js.bak:1]: I guess this is a copy of the old gulpfile.  Could you delete please.

File: core/components/overrides/govuk_elements/elements/_layout.scss [hunk 3]
```diff
@@ -145,7 +145,6 @@ section {
   width: 100%;
```
Comment (bobmeredith) [inline core/components/overrides/govuk_elements/elements/_layout.scss:145]: Nothing but reformatting in this file. Would you revert it please

File: core/components/react-components/webpack.config.js [hunk 4]
```diff
@@ -1,40 +1,40 @@
-const path = require('path');
```
Comment (bobmeredith) [inline core/components/react-components/webpack.config.js:1]: Looks like your tab width is different.  I guess there are no real changes to this file.  If not, please revert.

File: core/components/sass/components/utils/_base.scss [hunk 5]
```diff
@@ -7,57 +6,57 @@
 // Backgrounds
 
 $colours: (
-  ($black, black),
```
Comment (bobmeredith) [inline core/components/sass/components/utils/_base.scss:10]: Surely this looks wrong. These are sass variable declarations - they should start with '$'

File: core/components/sass/components/accordion/_base.scss [hunk 6]
```diff
@@ -14,7 +14,6 @@
   details.great-accordion:last-child {
```
Comment (bobmeredith) [inline core/components/sass/components/accordion/_base.scss:14]: This is just reformatting right? If so, best to revert.

File: core/components/sass/components/accordion/_base.scss [hunk 7]
```diff
@@ -47,8 +47,8 @@ details.great-accordion summary {
 }
 
 // For IE details-polyfill.js
-details.great-accordion[closed] summary:before,
```
Comment (bobmeredith) [inline core/components/sass/components/accordion/_base.scss:50]: This looks like a functional change right? the comma is lost.

File: core/components/static/stylesheets/elements-components.css [hunk 8]
```diff
@@ -34,8 +34,7 @@ details.great-accordion summary {
       details.great-accordion summary:after {
         font-size: 24px; } }
 
-details.great-accordion[closed] summary:before,
-details.great-accordion[open] summary:before {
+details.great-accordion[closed] summary:before details.great-accordion[open] summary:before {
```
Comment (bobmeredith) [inline core/components/static/stylesheets/elements-components.css:37]: ... and the change above is reflected in css here.

File: core/components/static/stylesheets/elements-components.css [hunk 9]
```diff
@@ -5204,29 +5202,30 @@ body:not(.js-enabled) .expandable-fieldset button.filter-collapse {
...
+      -webkit-box-orient: vertical;
+      -webkit-box-direction: reverse;
+      -ms-flex-direction: column-reverse;
+      flex-direction: column-reverse; } }
+  .shared-footer .magna-footer__great-logo img {
+    width: auto; }
 
 .shared-personalisation-bar {
   overflow-x: auto;
-  width: 100%;
```
Comment (bobmeredith) [inline core/components/static/stylesheets/elements-components.css:5229]: This is a real .css difference

File: core/components/static/stylesheets/elements-components.css [hunk 10]
```diff
@@ -5095,34 +5093,34 @@ body:not(.js-enabled) .expandable-fieldset button.filter-collapse {
...
           -webkit-transform: rotate(-45deg);
           transform: rotate(-45deg);
           margin-top: -12px; }
-    .shared-magna-header .header-links button .burger-icon {
+    .shared-magna-header .header-links #header-link-user-profile button .burger-icon {
       clear: right;
       content: "";
       float: right;
       height: 2px;
-      margin: 12px 0 10px 10px;
```
Comment (bobmeredith) [inline core/components/static/stylesheets/elements-components.css:5122]: This is a real .css difference

**Review by bobmeredith — CHANGES_REQUESTED** [review]
Hmm, In a perfect world, I would not expect the compiled .css to change at all, and yet I see a hodload of differences.  Some of them look functional and I don't see corresponding scss changes so I'm thinking something must be wrong. Also there are functional scss changes too - and there shouldn't be any of those.
Make sure it's clean and revert all scss file changes. Then build with the new scripts and you should see identical css and minified css I think. If that's the case, brilliant; if not, you'll need to work out if the changes are significant or not.

File: core/components/static/stylesheets/elements-components.css [hunk 11]
```diff
@@ -4724,509 +4724,3 @@ body:not(.js-enabled) .expandable-fieldset button.filter-collapse {
       width: calc(100vw - 8px);
       margin-left: calc(600px - 50vw);
       padding-left: calc((50vw - 600px) + (100% / 9 * 2) + 5px) !important; } }
-
-/* Shared styles across BAU and Magna pages */
-@font-face {
```
Comment (stevejalim) [inline core/components/static/stylesheets/elements-components.css:4729]: @seijihg Sorry, but how come this large amount of styling is disappearing? Are we sure it's safe to remove?

**Review by bobmeredith — APPROVED** [review]

**Pull request merged**